### PR TITLE
Update the development guide + add changelog script

### PR DESCRIPTION
**Pull request by cdrage**
**What kind of PR is this?**

/kind code-refactoring
/kind docs

**What does does this PR do / why we need it**:

- Adds changelog script to quickly generate a changelog using
github_changelog_generator

**Which issue(s) this PR fixes**:

Fixes N/A

**How to test changes / Special notes to the reviewer**:

N/A

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: scripts/changelog-script.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  echo -e "Must provide first and next release numbers..\nex: ./changelog-script.sh v1.0.0 v1.0.1"
+  exit 1
+fi
+
+github_changelog_generator \
+--user openshift \
+--project odo \
+-t $GITHUB_TOKEN \
+--since-tag v1.0.0 \
+--future-release v1.0.1 \
```
Comment (girishramnani) [inline scripts/changelog-script.sh]: Let this be user provided?

File: scripts/changelog-script.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+then
+  echo -e "Must provide first and next release numbers..\nex: ./changelog-script.sh v1.0.0 v1.0.1"
+  exit 1
+fi
+
+github_changelog_generator \
+--user openshift \
+--project odo \
+-t $GITHUB_TOKEN \
+--since-tag v1.0.0 \
```
Comment (girishramnani) [inline scripts/changelog-script.sh]: This could be taken from github releases as well via the api?

File: docs/dev/development.adoc [hunk 3]
```diff
@@ -448,15 +448,13 @@ $ git push upstream v0.0.1
...
 +
-NOTE: Do not upload any binaries for the release. When the new tag is created, Travis-CI starts a special deploy job. This job builds the binaries automatically (using `make prepare-release`) and then uploads it to the GitHub release page. When the job finishes you should see the binaries on the GitHub release page. The release is now marked as a draft.
 
 . Update the descriptions and publish the release.
 . Verify that packages have been uploaded to the `rpm` and `deb` repositories.
 . Update the Homebrew package:
 .. Check commit id for the released tag `git show-ref v0.0.1`
 .. Create a PR to update `:tag` and `:revision` in the https://github.com/kadel/homebrew-odo/blob/master/Formula/odo.rb[`odo.rb`] file
 in https://github.com/kadel/homebrew-odo[`kadel/homebrew-odo`].
-. Confirm the binaries are available in the GitHub release page.
```
Comment (mik-dass) [inline docs/dev/development.adoc]: I am a bit confused but don't we need to confirm that the binaries are there before the next step?

**Comment by cdrage** [comment]
I have updated the branch again and it's updated now.

File: docs/dev/development.adoc (hunk 3 above)
Comment (cdrage) [reply docs/dev/development.adoc]: Yeah, we no longer build binaries on travis / GitHub. It's all synced on the internal OpenShift CI.

File: docs/dev/development.adoc [hunk 4]
```diff
@@ -448,15 +443,17 @@ $ git push upstream v0.0.1
 ----
 *Or* create the new release using the GitHub site (this must be a proper release and not a draft).
 +
-NOTE: Do not upload any binaries for the release. When the new tag is created, Travis-CI starts a special deploy job. This job builds the binaries automatically (using `make prepare-release`) and then uploads it to the GitHub release page. When the job finishes you should see the binaries on the GitHub release page. The release is now marked as a draft.
 
-. Update the descriptions and publish the release.
+. Update the release description (changelog) on GitHub. To make things easier, a script was created to automatically generate a CHANGELOG and output to stdout.
++
+----
+$ ./scripts/generate-changelog.sh ${PREVIOUS_VERSION} ${NEW_VERSION}
```
Comment (amitkrout) [inline docs/dev/development.adoc]: +1

**Pull request merged**